### PR TITLE
[SOAR-17319] Salesforce Task Connection test & SDK Bump

### PR DESCRIPTION
**Pull request by rmurray-r7**
### Description

Task connection test added for Salesforce, alongside bumping the SDK and performing a plugin refresh

Describe the proposed changes:

  - SDK Bump to 6.1.0 & Plugin Refresh
  - Task Connection Test added

[Ticket](https://rapid7.atlassian.net/browse/SOAR-17319)
### Testing

Tested valid and invalid inputs for each field on Postman to ensure the correct exception handling is in place:

- Successful Test

image

- Invalid Login URL

image

-Invalid Client ID

image

- Invalid Secret Key

image

- Invalid Username & or Password

image

- Invalid Security Token

image

### CODE REVIEW

File: plugins/salesforce/komand_salesforce/util/api.py [hunk 1]
```diff
@@ -137,9 +137,9 @@ def query(self, query: str, next_page_id: str = None) -> dict:
         next_records_url = response.get("nextRecordsUrl")
         return {
             "records": response.get("records", []),
-            "next_page_id": next_records_url[next_records_url.index("query") + len("query") :]
```
Comment (rmurray-r7) [inline plugins/salesforce/komand_salesforce/util/api.py:140]: Black refactored the format

File: plugins/salesforce/komand_salesforce/connection/connection.py [hunk 2]
```diff
@@ -29,3 +34,57 @@ def test(self):
...
+                return_message += f"{endpoint} has passed \n"
+            except ApiException as error:
+                self.logger.info(error)
+                error_message += f"The connection test to Salesforce was unsuccessful \n The error was in the following test: {endpoint}. \nThis can be fixed by: {error.cause} \n"
+                self.logger.error(ConnectionTestException(cause=error.cause))
+                self.logger.error(ConnectionTestException(assistance=error.assistance))
+            except con_err as error:
+                self.logger.error(error)
+                error_message += f"The URL provided in the connection for the {endpoint} test is unreachable. Please ensure you are providing a valid URL when trying to connect."
+                self.logger.error(
```
Comment (joneill-r7) [inline plugins/salesforce/komand_salesforce/connection/connection.py:73]: looks like we've logged twice? do we need to create a ConnectionTestEception in this logger or just include the string?

File: plugins/salesforce/komand_salesforce/connection/connection.py [hunk 3]
```diff
@@ -29,3 +34,57 @@ def test(self):
...
+                        assistance=f"Please verify the URL specified for the {endpoint} test is valid and reachable before attempting to connect again."
+                    )
+                )
+
+        if not error_message:
+            self.logger.info(return_message)
+            return {"success": True}, return_message
+        else:
+            raise ConnectionTestException(
+                data=error_message,
```
Comment (joneill-r7) [inline plugins/salesforce/komand_salesforce/connection/connection.py:89]: think we should include the cause/assistance fields as well incase these get surfaced down the line

File: plugins/salesforce/komand_salesforce/connection/connection.py [hunk 4]
```diff
@@ -29,3 +34,57 @@ def test(self):
...
+                self.logger.info(f"Running test for {endpoint}")
+                return_message += f"Running test for {endpoint} \n"
+                method_execute, method_params = values[0], values[1]
+                _ = method_execute(method_params)
+                return_message += f"{endpoint} has passed \n"
+            except ApiException as error:
+                self.logger.info(error)
+                error_message += f"The connection test to Salesforce was unsuccessful \n The error was in the following test: {endpoint}. \nThis can be fixed by: {error.cause} \n"
+                self.logger.error(ConnectionTestException(cause=error.cause))
+                self.logger.error(ConnectionTestException(assistance=error.assistance))
```
Comment (joneill-r7) [inline plugins/salesforce/komand_salesforce/connection/connection.py:69]: Do these need to be the exception or can just be the string?

**Review by joneill-r7 — COMMENTED** [review]
Looks good - just a few questions around multiple loggers if we need them or can be included as one?

**Pull request merged**